### PR TITLE
Fixed test issues for dualtor topologies.

### PR DESCRIPTION
**Pull request by vivekverma-arista**
### Description of PR

Summary: Fixed test issue in dualtor topology.
Fixes # (issue): https://github.com/aristanetworks/sonic-qual.msft/issues/69

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [x] 202305
- [x] 202311

#### What is the motivation for this PR?
1. In some test cases the packets go to unselected ToR in case of active-active links or forwarded by the unselected ToR in case of active-standby links.
2. Some tests verify queue counters, but this check fails for UC1 in presence of active active links as UC1 always has gRPC traffic flowing through it.

#### How did you do it?
1. First issue can be addressed by using the following fixtures in case of active-standby (wherever missing, commit 1) :-
     a) toggle_all_simulator_ports_to_rand_selected_tor
     b) toggle_all_simulator_ports_to_enum_rand_one_per_hwsku_frontend_host_m
   
   To handle this in case of active-active links new fixtures are introduced (commit 3 & 4)
     a) setup_standby_ports_on_rand_unselected_tor
     b) setup_standby_ports_on_non_enum_rand_one_per_hwsku_frontend_host_m
     c) setup_standby_ports_on_rand_unselected_tor_unconditionally
     d) setup_standby_ports_on_non_enum_rand_one_per_hwsku_frontend_host_m_unconditionally
   
2. Second issue can be addressed by simply skipping the check for UC1 in presence of active-active links. (commit 2)

#### How did you verify/test it?
Verified on Arista 7260 device using dualtor/dualtor-aa topology.

### CODE REVIEW

File: tests/common/dualtor/dual_tor_utils.py [hunk 1]
```diff
@@ -1729,3 +1729,66 @@ def log_flap_counter(flap_counters):
         if unexpected_flap_mux_ports:
             logging.error(error_str)
             raise ValueError(error_str)
+
+
+@pytest.fixture
```
[the remark this answers is not in the document]
Comment (vivekverma-arista) [reply tests/common/dualtor/dual_tor_utils.py:1743]: Fixed.

File: tests/bgp/test_bgp_queue.py [hunk 2]
```diff
@@ -70,6 +72,10 @@ def test_bgp_queues(duthosts, enum_frontend_dut_hostname, enum_asic_index, tbinf
                 for port in mg_facts['minigraph_portchannels'][ifname]['members']:
                     logger.info("PortChannel '{}' : port {}".format(ifname, port))
                     for q in range(0, 7):
+                        # In case of active-active ports (dualtor-aa/dualtor-mixed) there will always be
```
[the remark this answers is not in the document]
Comment (vivekverma-arista) [reply tests/bgp/test_bgp_queue.py]: I have reverted the change.

**Review by lolyu — APPROVED** [review]
LGTM

**Comment by StormLiangMS** [comment]
hi @vivekverma-arista  could I ask your help to validate these changes with T0/T1? to avoid regression. pls  verify with 202305 image also.

**Comment by vivekverma-arista** [comment]
> hi @vivekverma-arista could I ask your help to validate these changes with T0/T1? to avoid regression. pls verify with 202305 image also.

Hi @StormLiangMS, I have verified this on t0 and t1, this introduces no new regression.

**Comment by yaqiangz** [comment]
Hi @vivekverma-arista , current PR has conflicts with master branch

**Comment by vivekverma-arista** [comment]
> Hi @vivekverma-arista , current PR has conflicts with master branch

Merge conflicts have been resolved.

**Comment by vivekverma-arista** [comment]
Hi @StormLiangMS can we merge this PR?

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Pull request merged**